test/per_file_coverage.sh

```diff
@@ -15,7 +15,7 @@ declare -a KNOWN_LOW_COVERAGE=(
 "source/common/matcher:94.6"
 "source/common/network:94.4" # Flaky, `activateFileEvents`, `startSecureTransport` and `ioctl`, listener_socket do not always report LCOV
 "source/common/network/dns_resolver:91.4"  # A few lines of MacOS code not tested in linux scripts. Tested in MacOS scripts
-"source/common/protobuf:96.5"
+"source/common/protobuf:96.3"
```

Comment:
Can we find a way to keep the existing coverage threshold for this directory?

Reply:
Not a relevant one. Literally every new line of code that is reachable is covered. We could add some nonsense padding lines in covered areas, or add a test to something completely unrelated to this change, would be the only feasible ways to re-increase the coverage percentage.

OR, an alternative I considered, give the helper class a separate constructor to allow to construct it with invalid input fields, expose that constructor only for testing (but it would still be built for production) and use that to add test coverage of the unreachable lines by making them reachable. Pretty sure that's even worse than the other options.

Reply:
Fair enough. Coverage LGTM

Reply:
Ended up actually improving the coverage, as the final version of `deterministic_hash.cc` now has 100% coverage (the switch with unreachable paths was excised thanks to jmarantz pointing out that we don't actually have to sort a list in order to hash it ignoring order), and the intermediate/current version of `utility.cc` still has the old lines covered for the lifespan of a runtime guard.